server/svix-server/src/queue/redis.rs

```diff
@@ -588,6 +594,31 @@ impl TaskQueueReceive for RedisQueueConsumer {
     }
 }
 
+async fn clear_acked(pool: RedisPool, queue: &str) -> Result<()> {
+    let mut conn = ctx!(pool.get().await)?;
+
+    let xpending_resp: StreamPendingReply =
+        ctx!(conn.query_async(Cmd::xpending(queue, WORKERS_GROUP)).await)?;
+
+    match xpending_resp {
+        StreamPendingReply::Empty => Ok(()),
+        StreamPendingReply::Data(resp) => {
+            // Get the smallest unACKed task that's been read, and know that any task with an ID
+            // less than that has already been ACKed
+            let min = resp.start_id;
```

Comment:
I'd rather add a safety mechanism where this item ID will have to be at least one week old too, so we really don't delete anything recent.

Reply:
Maybe there's no need actually, I guess that XPENDING by definition gets us the oldest one.